api-test/src/test/java/org/openmrs/module/metadatasharing/ConceptNamesTest.java

```diff
@@ -247,14 +246,194 @@ public void runOnImportServerAfterImport() throws Exception {
 				Collection<ConceptName> names = concept.getNames();
 				Set<String> expectedNames = new HashSet<String>();
 				expectedNames.addAll(Arrays.asList("AA", "AAG","a", "b"));
-				for (ConceptName name : names) {
-					assertTrue(name.getName() + " missing", expectedNames.remove(name.getName()));
-				}
-				
+				assertNull(diffConceptNameSets(concept.getNames(), expectedNames));
+
 				ConceptName preferredName = concept.getPreferredName(Locale.ENGLISH);
 				Assert.assertEquals("b must be preferred", "b", preferredName.getName());
 			}
 			
 		});
 	}
+
+	@Test
+	public void shouldAllowChangingNameCapitalization() throws Exception {
+		final String conceptUuid = UUID.randomUUID().toString();
+
+		runShareTest(new ShareTestHelper() {
+
+			@Override
+			public List<?> prepareExportServer() throws Exception {
+				Concept concept = new Concept();
+				concept.setUuid(conceptUuid);
+
+				ConceptName preferredName = new ConceptName("Aaa", Locale.ENGLISH);
+				concept.setPreferredName(preferredName);
+
+				ConceptName synonym = new ConceptName("Baa", Locale.ENGLISH);
+				concept.addName(synonym);
+
+				ConceptName synonym2 = new ConceptName("Aag", Locale.GERMAN);
+				concept.addName(synonym2);
+
+				Context.getConceptService().saveConcept(concept);
+				return Arrays.asList(concept);
+			}
+
+			/**
+			 * @see org.openmrs.module.metadatasharing.ShareTestHelper#prepareImportServer()
+			 */
+			@Override
+			public void prepareImportServer() throws Exception {
+				Concept concept = new Concept();
+				concept.setUuid(conceptUuid);
+
+				ConceptName preferredName = new ConceptName("AAA", Locale.ENGLISH);
+				concept.setPreferredName(preferredName);
+
+				ConceptName synonym = new ConceptName("BAA", Locale.ENGLISH);
+				concept.addName(synonym);
+
+				ConceptName synonym2 = new ConceptName("AAG", Locale.GERMAN);
+				concept.addName(synonym2);
+
+				Context.getConceptService().saveConcept(concept);
+			}
+
+			/**
+			 * @see org.openmrs.module.metadatasharing.ShareTestHelper#runOnImportServerBeforeImport(org.openmrs.module.metadatasharing.wrapper.PackageImporter)
+			 */
+			@Override
+			public void runOnImportServerBeforeImport(PackageImporter importer) throws Exception {
+				for (ImportedItem importedItem: importer.getImportedItems(0)) {
+					if (importedItem.getIncoming() instanceof Concept) {
+						importedItem.setImportType(ImportType.PREFER_THEIRS);
+					}
+				}
+			}
+
+			@Override
+			public void runOnImportServerAfterImport() throws Exception {
+				Concept concept = Context.getConceptService().getConceptByUuid(conceptUuid);
+
+				Collection<ConceptName> names = concept.getNames();
+				Set<String> expectedNames = new HashSet<String>(Arrays.asList("Aaa", "Baa", "Aag"));
+				assertNull(diffConceptNameSets(concept.getNames(), expectedNames));
+			}
+
+		});
+	}
+
+	@Test
+	public void shouldIgnoreDuplicateNamesWithDifferentCapitalization() throws Exception {
```

Comment:
Would this test fail without the above changes in `ConceptMergeHandler.java`?

Reply:
Oh no, shoot, glad you checked. The other one does fail but this one doesn't. The `ConceptValidationException` that gets thrown in real life doesn't get thrown in this test environment (or if it does, it gets swallowed somewhere).

Should I mark it with `@Ignore` with a concise explanation? It is possible for me to imagine that at some point the underlying test infrastructure is fixed so that it behaves more realistically. Or should I just delete it, since it doesn't add any value right now?